lib/requester/requester.js

```diff
@@ -51,97 +71,125 @@ var requests = require('./request-wrapper'),
-            responseString = util.arrayBufferToString(resBody);
-        }
+inherits(Requester = function (trace, options) {
+    this.options = {
+        keepAlive: _.get(options, 'keepAlive', true),
 
-        // Calculate the time taken for us to get the response.
-        responseTime = Date.now() - startTime;
-
-        // This helps us to unify the information from XHR or Node calls.
-        responseJSON = util.jsonifyResponse(res, requestOptions, responseString);
-
-        // Pull out cookies from the cookie jar, and make them chrome compatible.
-        cookies = (cookieJar && _.isFunction(cookieJar.getCookies)) ?
-            _.transform(cookieJar.getCookies(requestOptions.url), function (acc, cookie) {
-                acc.push(toChromeCookie(cookie));
-            }, []) : [];
-
-        // Response in the SDK format
-        response = new sdk.Response({ // @todo get rid of jsonifyResponse
-            code: responseJSON.statusCode,
-            status: res && res.statusMessage,
-            header: _.transform(responseJSON.headers, transformMultiValueHeaders, []),
-            stream: resBody,
-            responseTime: responseTime
-        });
+        // Create a cookie jar if one is not provided
+        cookieJar: options && options.cookieJar || RequestCookieJar(),
 
-        // Insert the missing sent headers in the request object, so that they get bubbled up into the UI
-        _.forOwn(responseJSON.request && responseJSON.request.headers, function (value, key) {
-            // todo: this should also be done in the request utils, so that it's appropriately updated there,
-            // and there's a clear separation between what we added vs what Node added.
-            request.upsertHeader({
-                key: key,
-                value: value
+        timeout: _.get(options, 'timeout'),
+        strictSSL: _.get(options, 'strictSSL'),
+        sendBodyWithGetRequests: _.get(options, 'sendBodyWithGetRequests', false),
+        followRedirects: _.get(options, 'followRedirects', true)
+    };
+    this.trace = trace;
+    Requester.super_.call(this);
+}, Emitter);
+
+_.extend(Requester.prototype, /** @lends Requester.prototype */ {
+
+    /**
+     * Perform an HTTP request.
+     *
+     * @param {Item} item
+     * @param {String} sendId
+     * @param {Cursor} cursor
+     * @param {Function} callback
+     */
+    request: function (item, sendId, cursor, callback) {
+        var self = this,
+            request = item.request,
+            cookieJar = self.options.cookieJar,
+            requestOptions = util.getRequestOptions(request, self.options),
+            startTime = Date.now();
+
+        return requests(request, requestOptions, function (err, res, resBody) {
+            if (err) {
+                // bubble up http errors
+                // @todo - Should we send an empty sdk Response here?
+                self.emit(sendId, err, self.trace, undefined, request);
```

Comment:
Is sending events only limited to triggering `IO` callbacks?

Should we add types to these events just in case for otherwise?